### PR TITLE
Add support for dynamic objects to kubectl apply.

### PR DESCRIPTION
**Pull request by brendandburns**
Addresses #2318 (I think)

@yue9944882 this is a sketch, let me know if you think it looks ok and I will add tests.

### CODE REVIEW

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/Kubectl.java [hunk 1]
```diff
@@ -308,6 +311,26 @@ protected GenericKubernetesApi<ApiType, KubernetesListObject> getGenericApi()
...
     }
+
+    protected GenericKubernetesApi<? extends KubernetesObject, ? extends KubernetesListObject> getGenericApi(KubernetesObject targetObj)
+        throws KubectlException {
+      if (targetObj instanceof DynamicKubernetesObject) {
+        DynamicKubernetesObject obj = (DynamicKubernetesObject)targetObj;
+        String gv = obj.getApiVersion();
+        int ix = gv.indexOf("/");
+        String group = ix == -1 ? "" : gv.substring(0, ix);
+        String version = gv.substring(ix + 1);
```
Comment (yue9944882) [inline extended/src/main/java/io/kubernetes/client/extended/kubectl/Kubectl.java:322]: https://github.com/kubernetes-client/java/blob/master/util/src/main/java/io/kubernetes/client/util/ModelMapper.java#L180-L190

i think we can promote the above code from `ModelMapper` to a utility helper function like `ParsedAPIVersion ModelMapper#parseAPIVersion(...)`, so we can reuse that here, wdyt?

Comment (brendandburns) [reply extended/src/main/java/io/kubernetes/client/extended/kubectl/Kubectl.java]: Done.

**Comment by brendandburns** [comment]
@yue9944882 updated per your comment. Please take another look. Once this looks good, I can add tests.

**Comment by yue9944882** [comment]
@brendandburns overall looks good, lets add some tests. (and fix the format)

**Comment by brendandburns** [comment]
Ok, I wrote tests and fixed formatting, this is ready for final review.

Thanks!

**Review by yue9944882 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**